examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,255 @@
+        points(i) = t;
+      });
+}
+
+// Dumps the data in a file in a csv format
+void dump_msl_data(std::string const &dump_filename,
+                   Kokkos::View<Point *, MemorySpace> const source_points,
+                   Kokkos::View<Point *, MemorySpace> const target_points,
+                   Kokkos::View<double *, MemorySpace> const source_values,
+                   Kokkos::View<double *, MemorySpace> const target_values,
+                   Kokkos::View<double *, MemorySpace> const approx_values)
+{
+  std::fstream dump_file_stream(dump_filename, std::ios::out | std::ios::trunc);
+  if (!dump_file_stream)
+    throw std::runtime_error("Unable to open/create file " + dump_filename);
+
+  for (int i = 0; i < DIM; i++)
+    dump_file_stream << "source coord " << i << ';';
+  dump_file_stream << "source value;";
+  for (int i = 0; i < DIM; i++)
+    dump_file_stream << "target coord " << i << ';';
+  dump_file_stream << "target value;approx value\n";
+
+  auto source_points_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, source_points);
+  auto target_points_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, target_points);
+  auto source_values_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, source_values);
+  auto target_values_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, target_values);
+  auto approx_values_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, approx_values);
+
+  int const num_points = source_points.extent(0);
+  for (int i = 0; i < num_points; i++)
+  {
+    for (int j = 0; j < DIM; j++)
+      dump_file_stream << source_points_host(i)[j] << ';';
+    dump_file_stream << source_values_host(i) << ';';
+    for (int j = 0; j < DIM; j++)
+      dump_file_stream << target_points_host(i)[j] << ';';
+    dump_file_stream << target_values_host(i) << ';' << approx_values_host(i)
+                     << '\n';
+  }
+}
+
+// Funtion to approximate
+KOKKOS_INLINE_FUNCTION double functionToApproximate(Point const &p)
+{
+  auto const x = p[0];
+  return (x > 0) ? 1 : 0;
+}
+
+void mls_example(std::size_t num_points, std::optional<int> num_neighbors,
+                 std::string const &dump_file)
+{
+  ExecutionSpace space{};
+  int const side_len = Kokkos::pow(num_points, 1. / DIM);
+
+  // Basis change
+  Kokkos::View<Point[DIM], MemorySpace> source_basis("Example::source_basis");
+  Kokkos::View<Point[DIM], MemorySpace> target_basis("Example::target_basis");
+  Kokkos::parallel_for(
+      "Example::fill_basis", Kokkos::RangePolicy<ExecutionSpace>(space, 0, 1),
+      KOKKOS_LAMBDA(int const) {
+        for (int i = 0; i < DIM; i++)
+        {
+          source_basis(i) = Point{};
+          target_basis(i) = Point{};
+          for (int j = 0; j < DIM; j++)
+          {
+            source_basis(i)[j] = double(i == j);
+            target_basis(i)[j] = double(i == j);
+          }
+        }
+      });
```

Comment:
Mostly to show that the user can do it. I also used it to build a skeded/triangular mesh in my report. It can be removed with the basis change function.